src/main/java/be/uantwerpen/minelabs/item/AbstractMagnet.java

```diff
@@ -0,0 +1,127 @@
+package be.uantwerpen.minelabs.item;
+
+import net.minecraft.client.item.TooltipContext;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.ExperienceOrbEntity;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.text.Text;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public abstract class AbstractMagnet extends Item {
+
+    protected static final float SPEED = 0.35F;
+    private static final String ENABLED_KEY = "minelabs.enabled";
+
+    public AbstractMagnet(Settings settings) {
+        super(settings);
+    }
+
+    @Override
+    public void inventoryTick(ItemStack stack, World world, Entity entity, int slot, boolean selected) {
+        // https://github.com/VazkiiMods/Botania/blob/1.19.x/Xplat/src/main/java/vazkii/botania/common/item/equipment/bauble/RingOfMagnetizationItem.java
+        // https://github.com/VazkiiMods/Botania/blob/9c94927a7289b8a1212ba38d1c3901bb16cc7ece/Xplat/src/main/java/vazkii/botania/common/helper/MathHelper.java#L30
+        // https://github.com/VazkiiMods/Botania/blob/9c94927a7289b8a1212ba38d1c3901bb16cc7ece/Xplat/src/main/java/vazkii/botania/common/helper/VecHelper.java
+        int RANGE = 8;
+        if (getState(stack)) {
+            if (entity instanceof PlayerEntity player) {
+                if (world.isClient) return;
+                List<Entity> toMove = player.getWorld().getOtherEntities(player,
+                        new Box(player.getX() - RANGE, player.getY() - RANGE, player.getZ() - RANGE,
+                                player.getX() + RANGE, player.getY() + RANGE, player.getZ() + RANGE),
+                        this::isAttractable);
+                for (Entity movable_entity : toMove) {
+                    double x = player.getX();
+                    double y = player.getY() + 0.75; // Mid-body / eye height
+                    double z = player.getZ();
+
+                    Vec3d vec = new Vec3d(x, y, z);
+
+                    Vec3d entityVector = new Vec3d(movable_entity.getX(), movable_entity.getY(), movable_entity.getZ());
+                    Vec3d finalVector = vec.subtract(entityVector);
+
+                    if (finalVector.length() > 1) {
+                        finalVector = finalVector.normalize();
+                    }
+
+                    //entity.setDeltaMovement(finalVector.scale(modifier));
+                    movable_entity.setVelocity(finalVector.multiply(SPEED));
```

Comment:
I wonder if we can get a nicer effect by adding to the velocity rather than setting it. That should give a smoother transition. Also set a max to the velocity otherwise @smtdm will have too much fun.

Reply:
Can be done. Want to see him break it before implementing a max though...